File: server/src/main/java/org/dan/jadalnia/app/order/pojo/OrderItem.java
package org.dan.jadalnia.app.order.pojo;

import lombok.Getter;
import lombok.Setter;
import org.dan.jadalnia.app.festival.menu.DishName;

import java.util.List;

@Getter
@Setter
public class OrderItem {
    private DishName name;
    private int quantity;
    private List<OrderItem> additions;
}

File: kot/src/main/kotlin/org/dan/jadalnia/app/order/pojo/OrderItem.kt
package org.dan.jadalnia.app.order.pojo;

import com.fasterxml.jackson.annotation.JsonCreator
import org.dan.jadalnia.app.festival.menu.DishName

data class OrderItem(
        val name: DishName,
        val quantity: Int,
        val additions: List<OrderItem>) {
    @JsonCreator constructor(
            name: DishName?,
            quantity: Int?,
            additions: List<OrderItem>?):
            this(
                    name ?: DishName("D"),
                    quantity ?: 1,
                    additions ?: emptyList<OrderItem>()) {
    }
}

